src/main/java/com/expediagroup/hiveberg/IcebergInputFormat.java

```diff
@@ -178,17 +191,22 @@ public float getProgress() throws IOException {
     }
   }
 
-  private static class IcebergSplit implements InputSplit {
+  private static class IcebergSplit extends FileSplit {
```

Comment:
Again, just checking, in addition to implementing all the DataIn and DataOut stuff we also still need to be of type `FileSplit`?

Reply:
Yeah this one was necessary too, because `HiveInputFormat` expects splits that are instances of FileSplit's

Reply:
OK, please add a comment for this too so we remember why we had to do all the crazy.